pinot-common/src/test/java/org/apache/pinot/common/metrics/BrokerJMXToPromMetricsTest.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.common.metrics;
+
+import com.yammer.metrics.core.MetricsRegistry;
+import com.yammer.metrics.reporting.JmxReporter;
+import io.prometheus.jmx.shaded.io.prometheus.client.exporter.HTTPServer;
+import java.net.URI;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.pinot.common.utils.SimpleHttpResponse;
+import org.apache.pinot.common.utils.http.HttpClient;
+import org.apache.pinot.plugin.metrics.yammer.YammerMetricsRegistry;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.apache.pinot.spi.metrics.PinotMetricUtils;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.apache.pinot.spi.utils.CommonConstants.CONFIG_OF_METRICS_FACTORY_CLASS_NAME;
+
+
+public class BrokerJMXToPromMetricsTest extends PinotJMXToPromMetricsTest {
+
+  private static final String EXPORTED_METRIC_PREFIX = "pinot_broker_";
+
+  private static final String EXPORTED_METRIC_PREFIX_EXCEPTIONS = "exceptions";
+
+  private BrokerMetrics _brokerMetrics;
+
+  private HTTPServer _httpServer;
+
+  @BeforeClass
+  public void setup()
+      throws Exception {
+
+    _httpServer = startExporter(PinotComponent.BROKER);
+
+    PinotConfiguration pinotConfiguration = new PinotConfiguration();
+    pinotConfiguration.setProperty(CONFIG_OF_METRICS_FACTORY_CLASS_NAME,
+        "org.apache.pinot.plugin.metrics.yammer.YammerMetricsFactory");
+    PinotMetricUtils.init(pinotConfiguration);
+
+    // Initialize ServerMetrics with the registry
+    YammerMetricsRegistry yammerMetricsRegistry = new YammerMetricsRegistry();
+    _brokerMetrics = new BrokerMetrics(yammerMetricsRegistry);
+
+    // Enable JMX reporting
+    MetricsRegistry metricsRegistry = (MetricsRegistry) yammerMetricsRegistry.getMetricsRegistry();
+    JmxReporter jmxReporter = new JmxReporter(metricsRegistry);
+    jmxReporter.start();
```

Comment:
Isn't this call already been called by `PinotMetricUtils.init(pinotConfiguration)` above?

Even if it isn't, we should be able to initialize the PinotMetricRegistry by instantiating `org.apache.pinot.plugin.metrics.yammer.YammerMetricsFactory` using reflection, then calling `init`. That returns the generic PinotMetricFactory and then we can call `PinotMetricUtils.makePinotJmxReporter`. But again, I think that is not actually needed.

Reply:
Made this configurable. Currently, I have added support for Yammer and Dropwizard. I can make it generic for any factory ideally - But do we really need to?